### DISCUSSION TITLE
How run Nitro on a different port?

### DISCUSSION THREAD

**Discussion by justindasilva** [nitrojs/nitro#192] [General]
So by default, when I run `nitro dev`, it runs on port `3000`. Is there anyway to specify that you want it to run on another port? I don't see anything in the documentation about port number.

**Comment by cpreston321** [comment]
@justindasilva 

It looks like it can be changed here: https://nitro.unjs.io/deploy/node.html#environment-variables

Happy Coding 😄

**Reply by justindasilva** [reply]
Hmm, this doesn't seem to work. I tried creating a `.env` file with `PORT = 4000` in it, and it still seems to create the server at `localhost:3000`. I tried `NITRO_PORT` as well.

**Reply by justindasilva** [reply]
Oh, sorry, I had to do a bit more research and I realized that I can edit my `package.json` to look like this:

```
 "scripts": {
    "dev": "PORT=4000 nitro dev",
    "build": "nitro build"
  },
```

I guess I just assumed that it would automatically take my `.env` file into account, but I probably need to install a module for that to work properly.

Either way, I was able to get it to work with the above, so I'm happy with that :)

**Reply by AntoninKabelka** [reply]
works !

**Reply by rs-duongnt1** [reply]
It's not working in windows. It throw error like: PORT is NOT reconized internal or external.

**Reply by TreasureUzoma** [reply]
I got mine to work on windows

```pnpm add cross-env```

```package.json
  "scripts": {
    "dev": "cross-env PORT=3005 nitro dev",
    "build": "nitro build"
  }
  ```

**Reply by colinmollenhour** [reply]
Now try overriding the websocket port.. I cannot for anything get rid of this:

>  ERROR  WebSocket server error: Port 24678 is already in use
